src/components/transactions/TransactionIcon.vue

```diff
@@ -15,6 +11,47 @@ export default {
     transactionGroup: {
       type: String,
       required: true
+    },
+    transactionType: {
+      type: String,
+      required: true
+    }
+  },
+  computed: {
+    icon() {
+      if (this.transactionType === `Sent`) {
```

Comment:
do you know ob the object switch approach?

```
const dictionary = {
  keyA: value,
  keyB: value
}

const valueForKey = dictionary[key]
```

Reply:
Definitely more cleaner but as I need to make some comparisons for Withdrawal, Vote and Deposit types like: 

`} else if (this.transactionType.search(`Voted`) === 0) {`

It's possible to do that in a dictionary?

Another option is just to simplify the caption for that messages and include the info in the tx detail,  the we can use a dictionary. Maybe a better solution.

Reply:
Looks funky to me. What are the options for `transactionType` that there are multiple with `Voted` in it?

Reply:
yeah I think you are on the right track. there should be an enum "transactionType" that should be different then the title.

Reply:
I finally implemented that way. The icon name should be the caption string, so that way we don't need a dictionary, only need to handle `Update withdraw address`, which is the only specific case. A bit hacky but simplier :-)

```javascript
computed: {
    icon() {
      if (this.transactionType.indexOf(` `) !== -1) {
        if (this.transactionType === `Update withdraw address`) {
          return `/img/icons/activity/Withdrawal.svg`
        } else {
          return `/img/icons/activity/${this.transactionType.substr(
            0,
            this.transactionType.indexOf(` `)
          )}.svg`
        }
      } else {
        return `/img/icons/activity/${this.transactionType}.svg`
      }
    }
  }
```

Reply:
I am still worried, that the transactionType is the title of the transaction. If we change the title now, we need to change the name of the icon. feels like an unnecessary linkage. But we can improve later.